appservice/src/main/java/nu/yona/server/subscriptions/rest/UserPhotoController.java

```diff
@@ -0,0 +1,151 @@
+/*******************************************************************************
+ * Copyright (c) 2015, 2017 Stichting Yona Foundation This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server.subscriptions.rest;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.imageio.ImageIO;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.ExposesResourceFor;
+import org.springframework.hateoas.Resource;
+import org.springframework.hateoas.mvc.ControllerLinkBuilder;
+import org.springframework.hateoas.mvc.ResourceAssemblerSupport;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.multipart.MultipartFile;
+
+import nu.yona.server.crypto.seckey.CryptoSession;
+import nu.yona.server.exceptions.InvalidDataException;
+import nu.yona.server.exceptions.YonaException;
+import nu.yona.server.rest.Constants;
+import nu.yona.server.subscriptions.rest.UserPhotoController.UserPhotoResource;
+import nu.yona.server.subscriptions.service.UserPhotoDto;
+import nu.yona.server.subscriptions.service.UserPhotoService;
+import nu.yona.server.subscriptions.service.UserService;
+
+@Controller
+@ExposesResourceFor(UserPhotoResource.class)
+@RequestMapping(value = "/", produces = { MediaType.APPLICATION_JSON_VALUE })
+public class UserPhotoController
+{
+	@Autowired
+	private UserService userService;
+
+	@Autowired
+	private UserPhotoService userPhotoService;
+
+	@RequestMapping(value = "/userPhotos/{userPhotoId}", method = RequestMethod.GET, produces = { MediaType.IMAGE_PNG_VALUE })
```

Comment:
I see you're always returning PNG here (see my comments on the Swagger spec). That's not an issue, I would think, but then the upload should not accept anything other than PNG.

Reply:
The upload converts to PNG, as discussed